lib/membrane_framerate_converter.ex

```diff
@@ -0,0 +1,135 @@
+defmodule Membrane.FramerateConverter do
+  @moduledoc """
+  Element converts video to target constant frame rate, by dropping and duplicating frames as necessary
+  Input video may have constant or variable frame rate.
+  Element expects each frame to be received in separate buffer.
+  Additionally, presentation timestamps must be passed in each buffer's metadata.
+  """
+  use Membrane.Filter
+  use Bunch
+
+  alias Membrane.Buffer
+  alias Membrane.Caps.Video.Raw
+
+  require Membrane.Logger
+
+  def_options framerate: [
+                spec: Tuple,
+                default: {30, 1},
+                description: """
+                Target framerate.
+                """
+              ]
+
+  def_output_pad :output,
+    caps: {Raw, aligned: true}
+
+  def_input_pad :input,
+    caps: {Raw, aligned: true},
+    demand_unit: :buffers
+
+  @impl true
+  def handle_init(%__MODULE__{} = options) do
+    state =
+      options
+      |> Map.from_struct()
+      |> Map.merge(%{
+        last_payload: nil,
+        last_metadata: nil,
+        target_pts: 0
+      })
+
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_demand(:output, size, :buffers, _ctx, state) do
+    {{:ok, demand: {:input, size}}, state}
+  end
+
+  @impl true
+  def handle_prepared_to_playing(_ctx, state) do
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_process(
+        :input,
+        %{payload: payload, metadata: metadata} = buffer,
+        _ctx,
+        %{last_payload: nil} = state
+      ) do
+    if not Map.has_key?(metadata, :pts), do: raise("Cannot cut stream without pts")
+
+    state = %{
+      state
+      | target_pts: metadata.pts,
+        last_metadata: metadata,
+        last_payload: payload
+    }
+
+    {{:ok, buffer: {:output, buffer}}, bump_target_pts(state)}
+  end
+
+  @impl true
+  def handle_process(
+        :input,
+        %{payload: payload, metadata: metadata},
+        _ctx,
+        %{last_payload: _last_payload} = state
+      ) do
+    if not Map.has_key?(metadata, :pts), do: raise("Cannot cut stream without pts")
+
+    {buffers, state} = create_new_frames(metadata, payload, state)
+    {{:ok, [buffer: {:output, buffers}, redemand: :output]}, state}
+  end
+
+  @impl true
+  def handle_caps(:input, caps, _context, %{framerate: framerate} = state) do
+    {{:ok, caps: {:output, %{caps | framerate: framerate}}, redemand: :output}, state}
+  end
+
+  defp bump_target_pts(state) do
+    use Ratio
+    %{target_pts: pts, framerate: {num, denom}} = state
+    target_pts = pts + Ratio.new(denom * Membrane.Time.second(), num)
+    %{state | target_pts: target_pts}
```

Comment:
I see that `Ratio` has its own operator `<|>`. What do you think about using it? 
```suggestion
    pts_difference = (denom * Membrane.Time.second()) <|> num
    %{state | target_pts: pts + pts_difference}
```

Reply:
This notation is taken from our H264 parser (which i linked in another comment). I personally also like the 'Ratio.new' notation, and it is safer, should any other module use '<|> ' operator.
Do you have some concerns about this notation?

Reply:
So we can leave `Ratio.new` for consistency with H264 Parser